path.c

```diff
@@ -1250,6 +1250,45 @@ char *strip_path_suffix(const char *path, const char *suffix)
 	return xstrndup(path, chomp_trailing_dir_sep(path, path_len));
 }
 
+int is_mount_point_via_stat(struct strbuf *path)
+{
+	size_t len = path->len;
+	unsigned int current_dev;
+	struct stat st;
+
+	if (!strcmp("/", path->buf))
+		return 1;
+
+	strbuf_addstr(path, "/.");
+	if (lstat(path->buf, &st)) {
+		/*
+		 * If we cannot access the current directory, we cannot say
+		 * that it is a bind mount.
+		 */
+		strbuf_setlen(path, len);
+		return 0;
+	}
+	current_dev = st.st_dev;
+
+	/* Now look at the current directoru */
```

Comment:
```suggestion
	/* Now look at the parent directory */
```

Reply:
I think it really is the current directory, not the parent directory, that we're looking at... but of course, it is a director*y*, not a director*u*... 😄

Reply:
The code obscures this a bit, but I am positive that this is the *parent* directory. We append '/.' to path above, then we append '.' here, which makes '/..'.

If the '/.' above is not needed, I suggest removing it. Maybe a 'dirname' method (which strips the tail component from a path which must not end on '/..') would make this more readable.